graphql_compiler/compiler/emit_match.py

```diff
@@ -123,10 +125,17 @@ def _represent_fold(fold_location, fold_ir_blocks):
     final_string = base_template % template_data
 
     for block in fold_ir_blocks:
-        if not isinstance(block, Filter):
-            raise AssertionError(u'Found a non-Filter IR block in the folded IR blocks: {} {} '
-                                 u'{}'.format(type(block), block, fold_ir_blocks))
-        final_string += u'[' + block.predicate.to_match() + u']'
+        if isinstance(block, Filter):
+            final_string += u'[' + block.predicate.to_match() + u']'
+        elif isinstance(block, Traverse):
+            template_data = {
+                'direction': block.direction,
+                'edge_name': block.edge_name,
+            }
+            final_string += traverse_edge_template % template_data
+        elif not isinstance(block, Backtrack):
```

Comment:
We probably shouldn't have `Backtrack` blocks here at all, this can probably be just an `else`

Reply:
That's what I thought, but `extract_folds_from_ir_blocks` does not remove the `Backtrack` blocks, and these end up in `fold_ir_blocks`. Should I modify `extract_folds_from_ir_blocks`?

Reply:
No -- `extract_folds_from_ir_blocks` is a common (backend-language-agnostic) lowering step, so it needs to keep `Backtrack` blocks. I'm not sure if they are needed in Gremlin, but for MATCH we know they aren't necessary within a `@fold`.

We don't want too much complexity all in one place and one function. We should have a separate lowering step for extracted folded blocks where the `Backtrack` blocks are removed. This can be a common step if Gremlin needs it too, or a MATCH-specific one if not.